From 15601b4f10f30eaeab2a33006ef00e9f0888da7c Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <rosslagerwall@gmail.com>
Date: Thu, 7 Nov 2013 12:11:40 +0200
Subject: [PATCH] sftp: Fix handling of multiple reads of the packet length

In certain cases, reading the packet length may take more than one call.
Make this work by calculating the offset into the reply_size correctly.

https://bugzilla.gnome.org/show_bug.cgi?id=532951
---
 daemon/gvfsbackendsftp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index d32ec69..23916b1 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -1242,7 +1242,7 @@ read_reply_async_got_len  (GObject *source_object,
   if (backend->reply_size_read < 4)
     {
       g_input_stream_read_async (backend->reply_stream,
-				 &backend->reply_size + backend->reply_size_read, 4 - backend->reply_size_read,
+				 (char *)&backend->reply_size + backend->reply_size_read, 4 - backend->reply_size_read,
 				 0, backend->reply_stream_cancellable, read_reply_async_got_len,
 				 backend);
       return;
-- 
1.9.0

